piksi_tools/console/console.py

```diff
@@ -834,24 +834,23 @@ def set_connection_mode(self, mode):
 def connection_from_args(args):
     cnx_data = None
     connect_error = None
-    retry_connection = True
     description = ''
     mode = ''
     port = getattr(args, 'port', None)
     baud = getattr(args, 'baud', None)
 
     if port is not None:
-        if args.tcp:
+        if args.tcp is True:
```

Comment:
Sorry, should have explained this more, the `is None` and `is not None` idiom usually only applies to the None value (which is a singleton in Python) and is basically done because it avoids any equality operators that an object might implement and forces Python to evaluate the question "is this object identical to the None singleton" instead of "call the equality operator on this object and pass None".

So, I may've given misleading advice on the idiom early... but if you want to check the "truthiness" of something, just using `if blah` is fine (None, False, empty lists/sets will all evaluate to false).  However, if you explicitly want to know if something is undefined/None (for example, to initialize the value because the argument wasn't specified) then `is None` or `is not None` is more appropriate.

Reply:
gotcha, basically I am using `getattr` to avoid exceptions (as opposed to using a `try except` for each) if `args` doesn't have the common set of values defined. I'm sure this is non-idiomatic as well *shrug*